### PR TITLE
APS-570 Remove AP Area 'National'

### PR DESCRIPTION
**Pull request by davidatkinsuk**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt [hunk 1]
```diff
@@ -34,7 +34,7 @@ import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.UserWorkload
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.community.StaffProbationArea
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.community.StaffUserDetails
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.AuthorisableActionResult
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas1.Cas1UserMappingService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas1.Cas1ApAreaMappingService
```
Comment (edavey) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt:37]: 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1ApAreaMappingService.kt [hunk 2]
```diff
@@ -47,9 +47,9 @@ class Cas1ApAreaMappingService(
     usersProbationRegion: ProbationRegionEntity,
     deliusUser: StaffUserDetails,
   ): ApAreaEntity {
-    val deliusProbationAreaCode = deliusUser.probationArea.code
-    val isInNationalProbationArea = listOf("N43", "N41", "XXX").contains(deliusProbationAreaCode)
-    if (isInNationalProbationArea) {
+    if (usersProbationRegion.apArea != null) {
```
Comment (edavey) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1ApAreaMappingService.kt:50]: nice! -- thanks for spotting and using this opportunity to remove these magic probation area codes.

File: src/main/resources/db/migration/all/20240716095118__remove_national_ap_area.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+DELETE FROM ap_areas where identifier = 'National';
```
Comment (edavey) [inline src/main/resources/db/migration/all/20240716095118__remove_national_ap_area.sql:1]: I love it when a plan comes together 😎

**Review by edavey — APPROVED** [review]
Excellent, thanks for getting to the bottom of this.

File: src/main/resources/db/migration/all/20240716095118__remove_national_ap_area.sql (hunk 3 above)
Comment (davidatkinsuk) [reply src/main/resources/db/migration/all/20240716095118__remove_national_ap_area.sql:1]: So much work had to go into enabling that one little line. Thanks for your help working all of this out!

**Pull request merged**